lib/pcr.c

```diff
@@ -191,7 +191,7 @@ bool pcr_print_pcr_struct(TPML_PCR_SELECTION *pcr_select, tpm2_pcrs *pcrs) {
         tpm2_tool_output("  %s:\n", alg_name);
 
         // Loop through all PCRs in this bank
-        UINT8 pcr_id;
+        int pcr_id;
```

Comment:
I think the better fix would be to force everything to unsigned. I think the multiplication is going signed since the literal will int in an int, C uses int as it's type and thus we're getting signed results from it. I think changing the 8 to 8u will fix the issue and prevent us from undefined overflow on the counter variable.

Reply:
If I understand the integer promotion rules correctly, using `8u` instead of `8` will still convert the upper bound to `unsigned int`, let me run a quick test with LGTM on this PR. Using `8u` in combination with changing `pcr_id` to `unsigned int` seems reasonable.

Reply:
Yeah, using only `8u` [does not fix these three alerts](https://github.com/tpm2-software/tpm2-tools/pull/1856#issuecomment-558713772). Changing both `8`→`8u` and `UINT8`→`unsigned int` should hopefully bring it back to fixing all four alerts.

Reply:
Indeed, we are [back to normal](https://github.com/tpm2-software/tpm2-tools/pull/1856#issuecomment-558718538).